src/providers/aws.rs

```diff
@@ -1321,6 +1304,31 @@ impl RegionLauncher {
             .instrument(key_span)
             .await;
         }
+
+        if !self.security_group_id.trim().is_empty() {
+            let group_span = tracing::trace_span!("security_group", id = %self.security_group_id);
+            async {
+                tracing::trace!("removing security group.");
+                // clean up security groups and keys
+                loop {
+                    let mut req = rusoto_ec2::DeleteSecurityGroupRequest::default();
+                    req.group_id = Some(self.security_group_id.clone());
+                    if let Err(e) = client.delete_security_group(req).in_current_span().await {
```

Comment:
The `in_current_span` here actually isn't necessary. Just pushed a commit to `master` that removes this and some other extraneous ones.